### DISCUSSION TITLE
赤外線発光、受光がほしい

### DISCUSSION THREAD

**Discussion by techan2007** [algyan/XIAOGYAN#12] [Ideas]
ども。のぐちです。

なにやら面白そうなので覗いています。
XAIOつかうのですね。

ぼくの周りでもいろいろやってみたい、というひとは
ちらほらいるんですが、赤外線リモコンの需要は結構あるようです。
で、赤外線LEDと赤外線受光センサがつけられるとうれしいかな、と。

赤外線LED
https://akizukidenshi.com/catalog/g/gI-03261/
受光センサ
https://akizukidenshi.com/catalog/g/gI-06487/

あたりが使いやすいのかな、と。
そういう用途に使えるように
D0 - SPEAKER
D3 - SPEAKER-ENABLE
のピンをジャンパ経由で外部に引き出せるようにしてもらえれば
なぁ、とか。

R5の手前とR6のあとでか、カットパターン(JP？)で切り分けて
VCC,GND,D0,D3
の４ピンを引き出せるようにしてもらえるといいのかも。

せっかくクラウドも視野にいれてるなら、スマートリモコン
みたいなのができるのは、たのしいかもしれず。

勝手なことばかりですみませんが、いいものができればいいなぁ、と。

**Comment by matsujirushi** [comment]
@techan2007 さん、アイデアありがとうございます！

赤外線は、まったく気づいていませんでした。
発光だけで十分かなーと思ったのですが、受光はどんな使い方（遊び方）が考えられるでしょうか？

**Reply by techan2007** [reply]
受光がないと、学習リモコンがつくれないから。
ある程度、わかった波形をそれなりに出すならいいけれど
それでは自分の使いたいリモコンが対応してるか、わからないし

そこまで、自分で学習させて、自分で使える、ってのがうれしいのかな、と。
発光だけならオンボードのLEDのパッドからうばっちゃえば、まぁ済むんだけど
面白くないよね

**Reply by matsujirushi** [reply]
なるほど、学習用に使うんですね。

**Reply by techan2007** [reply]
最近はIrLEDのドライバつきのやつはあんまりないのね。
38kHzで変調して出さないとダメで、数年まえはシャープのデバイスがあったんだけど。

一昨年くらいにPSoCでやったんだけど、あのときは内蔵のCLKで簡単に変調出力できたけれど
XIAOだとめんどくさいか。。。

**Comment by matsujirushi** [comment]
@techan2007 さん、

XIAOピン不足のため、割り当てできずorz

ES2基板では、左側にユニバーサルの場所を用意しました。ここに赤外線LEDと赤外線受光センサを載せて、GPIO(SCL, SDA)制御するということでどうでしょうか？ ( https://github.com/algyan/algyan_8th/discussions/18#discussioncomment-4618862 )
（その先は、、、良い感じのものが出来てから、Groveモジュールの形で作って、左側にマウントするのが良さそう。）

**Reply by techan2007** [reply]
いいッス（笑）
赤外線発光のときに38kHz変調を載せないとだめなんだけど、
昔あった変調機能付IrLEDが秋月からなくなってる。
となると、PWM出力と外部回路で合成するかなにか
しないと駄目なので。
I2CスレーブのPSoCつくっておいちゃうかなー

**Reply by matsujirushi** [reply]
作っておいちゃってくださいーｗ
そういえば、何かは忘れましたが38kHzじゃない家電も存在しますよね？エアコンだったかな。
周波数を設定できると良いかも。
